packages/openid4vc-holder/src/issuance/OpenId4VciHolderService.ts

```diff
@@ -0,0 +1,782 @@
+
+      if (credentialWithMetadata.offerType === OfferedCredentialType.InlineCredentialOffer) {
+        // Inline Credential Offers come with no context
+      } else {
+        if ('@context' in credentialWithMetadata.credentialSupported) {
+          context = credentialWithMetadata.credentialSupported['@context'] as unknown as string
+        } else {
+          throw new AriesFrameworkError('Could not find @context in credentialSupported.')
+        }
+      }
+
+      return {
+        type,
+        format,
+        types,
+        locations,
+        '@context': context,
+      }
+    } else {
+      throw new AriesFrameworkError(`Cannot create authorization_details. Unsupported credential format ${format}.`)
+    }
+  }
+
+  public async resolveAuthorizationRequest(
+    agentContext: AgentContext,
+    resolvedCredentialOffer: ResolvedCredentialOffer,
+    authCodeFlowOptions: AuthCodeFlowOptions
+  ): Promise<ResolvedAuthorizationRequest> {
+    const { credentialOfferPayload, metadata: _metadata, version } = resolvedCredentialOffer
+    const codeVerifier = (
+      await Promise.all([agentContext.wallet.generateNonce(), agentContext.wallet.generateNonce()])
+    ).join('')
+    const codeVerifierSha256 = Hasher.hash(TypedArrayEncoder.fromString(codeVerifier), 'sha2-256')
+    const codeChallenge = TypedArrayEncoder.toBase64URL(codeVerifierSha256)
+
+    const { metadata, issuerMetadata } = await getMetadataFromCredentialOffer(credentialOfferPayload, _metadata)
+
+    const offeredCredentialsWithMetadata = getOfferedCredentialsWithMetadata(
+      credentialOfferPayload,
+      issuerMetadata,
+      version
+    )
+
+    let authDetailsLocation: string | undefined
+    if (issuerMetadata.authorization_server) {
+      authDetailsLocation = metadata.issuer
+    }
+
+    const authDetails = offeredCredentialsWithMetadata.map((credential) =>
+      this.getAuthDetailsFromOfferedCredential(credential, authDetailsLocation, version)
+    )
+
+    this.logger.debug('Converted code_verifier to code_challenge', {
+      codeVerifier: codeVerifier,
+      sha256: codeVerifierSha256.toString(),
+      base64Url: codeChallenge,
+    })
+
+    const { clientId, redirectUri, scope } = authCodeFlowOptions
+    const authorizationRequestUri = await createAuthorizationRequestUri({
+      credentialOffer: credentialOfferPayload,
+      clientId,
+      codeChallengeMethod: CodeChallengeMethod.SHA256,
+      codeChallenge,
+      redirectUri,
+      scope,
+      authDetails,
+      metadata,
+    })
+
+    return {
+      ...authCodeFlowOptions,
+      codeVerifier,
+      authorizationRequestUri,
+    }
+  }
+
+  public async acceptCredentialOffer(
+    agentContext: AgentContext,
+    options: {
+      resolvedCredentialOffer: ResolvedCredentialOffer
+      acceptCredentialOfferOptions: AcceptCredentialOfferOptions
+      resolvedAuthorizationRequest?: ResolvedAuthorizationRequest & { code: string }
```

Comment:
Can't we add the `{code:string}` to the `ResolvedAuthorizationRequest`? Or create a new type?